stingray/lightcurve.py

```diff
@@ -664,38 +770,32 @@ def rebin(self, dt_new=None, f=None, method='sum'):
             raise ValueError("New time resolution must be larger than "
                              "old time resolution!")
 
-        if self.gti is None:
-
-            bin_time, bin_counts, bin_err, _ = \
-                utils.rebin_data(self.time, self.counts, dt_new,
-                                 yerr=self.counts_err, method=method)
-
-        else:
-            bin_time, bin_counts, bin_err = [], [], []
-            gti_new = []
-            for g in self.gti:
-                if g[1] - g[0] < dt_new:
-                    continue
-                else:
-                    # find start and end of GTI segment in data
-                    start_ind = self.time.searchsorted(g[0])
-                    end_ind = self.time.searchsorted(g[1])
+        bin_time, bin_counts, bin_err = [], [], []
+        gti_new = []
+        for g in self.gti:
```

Comment:
I think this line will fail if `gti is None`? Can we be sure that `gti is None` is never true?

Reply:
`__init__` always generates gtis. Is there a use case where the user sets gtis to None?

Reply:
... probably not? :)